Apps/GatewayApi/src/Services/DelegationService.cs

```diff
@@ -16,57 +16,101 @@
         private readonly IConfiguration configuration;
         private readonly IMapper autoMapper;
         private readonly ILogger<DelegationService> logger;
+        private readonly IDataProtector dataProtector;
         private readonly IDelegationDelegate delegationDelegate;
-        private readonly IPatientDetailsService patientDetailsService;
         private readonly IHashDelegate hashDelegate;
+        private readonly IEmailQueueService emailQueueService;
+        private readonly IPatientDetailsService patientDetailsService;
+        private readonly EmailTemplateConfig emailTemplateConfig;
+        private readonly DelegationInviteConfig delegationInviteConfig;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="DelegationService"/> class.
         /// </summary>
         /// <param name="configuration">The configuration to use.</param>
         /// <param name="autoMapper">The injected auto mapper provider.</param>
         /// <param name="logger">The injected logger.</param>
+        /// <param name="dataProtectionProvider">The injected data protection provider.</param>
         /// <param name="delegationDelegate">The injected delegation delegate.</param>
         /// <param name="hashDelegate">The injected hash delegate.</param>
+        /// <param name="emailQueueService">The injected email queue service.</param>
         /// <param name="patientDetailsService">The injected patient details service.</param>
         public DelegationService(
             IConfiguration configuration,
             IMapper autoMapper,
             ILogger<DelegationService> logger,
+            IDataProtectionProvider dataProtectionProvider,
             IDelegationDelegate delegationDelegate,
             IHashDelegate hashDelegate,
+            IEmailQueueService emailQueueService,
             IPatientDetailsService patientDetailsService)
         {
             this.configuration = configuration;
             this.autoMapper = autoMapper;
             this.logger = logger;
+            this.dataProtector = dataProtectionProvider.CreateProtector(DataProtectionPurpose);
             this.delegationDelegate = delegationDelegate;
             this.hashDelegate = hashDelegate;
+            this.emailQueueService = emailQueueService;
             this.patientDetailsService = patientDetailsService;
+            this.emailTemplateConfig = configuration.GetSection(EmailTemplateConfig.ConfigurationSectionKey).Get<EmailTemplateConfig>() ?? new();
+            this.delegationInviteConfig = configuration.GetSection(DelegationInviteConfig.ConfigurationSectionKey).Get<DelegationInviteConfig>() ?? new();
+        }
+
+        /// <inheritdoc/>
+        public async Task AssociateDelegationAsync(string delegateHdid, string encryptedDelegationId, CancellationToken ct = default)
+        {
+            ValidationResult validationResult = await new AssociateDelegationRequestValidator(encryptedDelegationId).ValidateAsync(ct, ct);
```

Comment:
`ct` is being passed as the object to validate.  Earlier suggestion would change this call to:
```
AssociateDelegationRequestValidator().ValidateAsync(encryptedDelegationId, ct)
```

Reply:
Changed.